src/main/java/task/TaskManager.java

```diff
@@ -0,0 +1,164 @@
+        try {
+            int taskNumber = InputParser.getTaskNumber(commandComponents);
+            allTasks.get(taskNumber).setTaskCompleted();
+            Display.displayTaskCompleted(allTasks.get(taskNumber).getTask());
+        } catch (IndexOutOfBoundsException e) {
+            Error.displayTaskNonExistentError();
+        } catch (NumberFormatException e) {
+            Error.displayNotANumberError();
+        }
+    }
+
+    public void listTask() {
+        Display.printListTaskLine();
+        for (int i = 0; i < taskCount; i++) {
+            System.out.println(i + INDEX_OFFSET + ". " + allTasks.get(i));
+        }
+        Display.printListTaskLine();
+    }
+
+    public void findTask(String taskKeyword) {
+        Display.printListTaskLine();
+        allTasks.stream()
+                .filter((task) -> task.getTask().contains(taskKeyword))
+                .forEach(System.out::println);
+        Display.printListTaskLine();
+    }
+
+    public void addSavedTodoTask(Boolean isCompleted, String taskDetails) {
+        try {
+            allTasks.add(new Todo(InputParser.getTaskName(taskDetails)));
+            if (isCompleted) {
+                allTasks.get(taskCount).setTaskCompleted();
+            }
+            increaseTaskCount();
+        } catch (DukeTaskNameEmptyException e) {
+            Error.displayFileSavedTaskNameEmptyError();
+        }
+    }
+
+    public void addSavedDeadlineTask(Boolean isCompleted, String taskDetails) {
+        try {
+            allTasks.add(new Deadline(InputParser.getTaskNameComponent(taskDetails),
+                    InputParser.getSavedDateComponent(taskDetails),
+                    InputParser.getSavedTimeComponent(taskDetails)));
+            if (isCompleted) {
+                allTasks.get(taskCount).setTaskCompleted();
+            }
+            increaseTaskCount();
+        } catch (IndexOutOfBoundsException e) {
+            Error.displayFileSavedTaskFormatError();
+        } catch (DukeTaskNameEmptyException e) {
+            Error.displayFileSavedTaskNameEmptyError();
+        } catch (DateTimeParseException e) {
+            Error.displayFileSavedDateFormatError();
+        }
+    }
+
+    public void addSavedEventTask(Boolean isCompleted, String taskDetails) {
+        try {
+            allTasks.add(new Event(InputParser.getTaskNameComponent(taskDetails),
+                    InputParser.getSavedDateComponent(taskDetails),
+                    InputParser.getSavedTimeComponent(taskDetails)));
+            if (isCompleted) {
+                allTasks.get(taskCount).setTaskCompleted();
+            }
+            increaseTaskCount();
+        } catch (IndexOutOfBoundsException e) {
+            Error.displayFileSavedTaskFormatError();
+        } catch (DukeTaskNameEmptyException e) {
+            Error.displayFileSavedTaskNameEmptyError();
+        } catch (DateTimeParseException e) {
+            Error.displayFileSavedDateFormatError();
+        }

```

Comment:
repeated code here (same from 137 to 144), perhaps consider SLAP?

Reply:
Thank you for pointing it out!